app/controllers/api/v1/admin/trends/links_controller.rb

```diff
@@ -1,7 +1,38 @@
 # frozen_string_literal: true
 
 class Api::V1::Admin::Trends::LinksController < Api::V1::Trends::LinksController
-  before_action -> { authorize_if_got_token! :'admin:read' }
+  include Authorization
+
+  before_action -> { authorize_if_got_token! :'admin:read' }, only: :index
+  before_action -> { authorize_if_got_token! :'admin:write' }, except: :index
+
+  after_action :verify_authorized
+
+  def index
+    authorize :preview_card, :index?
```

Comment:
This actually changes the behavior of that endpoint, and is kind of inconsistent with the remaining of the method: indeed, `authorize :preview_card, :index?` is equivalent to checking `current_user&.can?(:manage_taxonomies)` and will forbid accessing this endpoint without that permission, rather than returning a list of actually trending links with no additional information.

Reply:
You're right. I got a little overzealous writing the tests for these new methods. I have fixed it.